runtime/parachains/src/paras_inherent.rs

```diff
@@ -749,21 +761,7 @@ fn apply_weight_limit<T: Config + inclusion::Config>(
 			random_sel::<BackedCandidate<<T as frame_system::Config>::Hash>, _>(
 				&mut rng,
 				candidates.clone(),
-				|backed_candidate| {
-					let code_weight = backed_candidate
-						.candidate
-						.commitments
-						.new_validation_code
-						.as_ref()
-						.map(|_code| {
-							<<T as Config>::WeightInfo as WeightInfo>::enter_backed_candidate_code_upgrade()
-						})
-						.unwrap_or(0);
-					<<T as Config>::WeightInfo as WeightInfo>::enter_backed_candidates_variable(
-						backed_candidate.validity_votes.len() as u32,
-					)
-					.saturating_add(code_weight)
```

Comment:
also, we should max code_weight and output of `enter_backed_candidates_variable` since code_weight assumes worst case validity votes anyways